universal-login-react/src/ui/Notifications/EmojiForm.tsx

```diff
@@ -30,10 +30,14 @@ export const EmojiForm = ({deployedWallet, hideTitle, className, onDenyRequests,
   const [gasParameters, setGasParameters] = useState<GasParameters | undefined>(undefined);
 
   const [notifications, setNotifications] = useState<Notification[]>([]);
-  useEffect(() => deployedWallet.subscribeAuthorisations(setNotifications), []);
+  const [soleAddress, setSoleAddress] = useState<string | undefined>(undefined);
 
-  const addresses = filterNotificationByCodePrefix(notifications, enteredCode);
-  const soleAddress = addresses.length === 1 ? addresses[0] : undefined;
+  useEffect(() => deployedWallet.subscribeAuthorisations(onNotificationsChange), []);
+
+  const onNotificationsChange = (notifications: Notification[]) => {
+    setNotifications(notifications);
+    updateSoleAddress(filterNotificationByCodePrefix(notifications, enteredCode));
```

Comment:
Instead of `onNotificationsChange` and `updateEnteredCode` we can do 
```
useEffect(() => {
  updateSoleAddress(filterNotificationByCodePrefix(notifications, code))
}, [notification, code])
```

And probably inline `updateSoleAddress`